lib/src/simple_module.dart

```diff
@@ -0,0 +1,67 @@
+library w_module.src.simple_module;
+
+/// A [SimpleModule] encapsulates a well-scoped logical unit of functionality and
+/// exposes a discrete public interface for consumers.
+///
+/// The public interface of a [SimpleModule] is comprised of [api], [events],
+/// and [components]:
+/// - The [api] class exposes public methods that can be used to mutate or query
+///   module data.
+/// - The [events] class exposes streams that can be listened to for
+///   notification of internal module state change.
+/// - The [components] class exposes react-dart compatible UI components that
+///   can be used to render module data.
+
+abstract class SimpleModule {
```

Comment:
#nit: you could remove the blank line between the doc comment and the class declaration.